src/compile/mark/init.ts

```diff
@@ -25,6 +25,10 @@ export function normalizeMarkDef(mark: Mark | MarkDef, encoding: Encoding<string
     markDef.opacity = opacity(markDef.type, encoding);
   }
 
+  if (specifiedOpacity === 1.0) {
```

Comment:
This seems like a pretty dangerous condition to add.  Can you explain why?

Reply:
In the transform extraction, we need to set the opacity to avoid auto opacity of 0.7. However, in this case the Vega specs look different from the original that had a default of no opacity defined. The Vega default is 1 so this is safe to do.